xarray/tests/test_duck_array_ops.py

```diff
@@ -373,6 +373,16 @@ def test_cftime_datetime_mean_dask_error():
         da.mean()
 
 
+def test_mean_dtype():
+    ds = Dataset()
+    ds["pos"] = [1, 2, 3]
+    ds["data"] = ("pos", "time"), [[1.0, 2.0], [3.0, 4.0], [5.0, 6.0]]
+    ds["var"] = "pos", [2, 3, 4]
```

Comment:
FYI you can construct the dataset in a single expression, but it's also fine as-is (`Dataset(dict(pos=..., data=...))`)